lib/cog/queries/user.ex

```diff
@@ -9,6 +9,14 @@ defmodule Cog.Queries.User do
     preload: [:chat_handles]
   end
 
+  def for_chat_provider_user_id(chat_provider_user_id, provider) when is_binary(provider) do
```

Comment:
We've now got 3 queries in this module that use the same from and join clauses (as well as some of the where and preloads, too!) Any way we can consolidate / extract the common bits and use composable queries?